### PR TITLE
Fix UCAS matches summary in support

### PR DESCRIPTION
**Pull request by AgaDufrat**
## Context

We are getting errors for individual UCAS match pages: 
https://sentry.io/organizations/dfe-bat/issues/1936503787/?project=1765973

Screenshot 2020-10-06 at 09 03 28

## Changes proposed in this pull request

**- Consider recruitment cycle when looking up course for UCAS matches**
Now that we are approaching the next recruitment cycle we have two courses with the same code and provider. One for each recruitment cycle year. Decided to hardcore it instead of using `RecruitmentCycle.current_year`, `RecruitmentCycle.previous_year` as it’s going to change soon but we still want to compare courses in 2020 recruitment cycle. 

**- Fix UCAS matches summary for apply again candidates**
When a candidate applied again they would have two application forms. `find_by` was returning the latest application form, which in most cases was incomplete and not sent to providers and therefore had no application choices. Candidate’s ids for which this was the case: 96, 471, 3410, 1030, 2485, 2850, 3327, 3616.

## Guidance to review

Review questions:
- thought on hardcoding recruitment cycle year
- is there a fancy join I can do to avoid 2 queries in the 2nd commit?

Don't think we can reproduce the same issue locally. We can get the same error by running `GenerateTestData.new(10).generate` a few times but this is because some test application forms are generated without application choices, _I think._

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/models/ucas_matched_application.rb [hunk 1]
```diff
@@ -1,10 +1,16 @@
 class UCASMatchedApplication
+  RECRUITMENT_CYCLE_YEAR_FOR_UCAS_MATCHES = 2020
```
Comment (duncanjbrown) [inline app/models/ucas_matched_application.rb:2]: What if we looked at `created_at` on the match, and checked which cycle bucket it fell into? I don't think any new matches should be created after the cycle closes.

File: app/models/ucas_matched_application.rb [hunk 2]
```diff
@@ -33,9 +33,10 @@ def status
     if ucas_scheme?
       mapped_ucas_status
     else
-      ApplicationForm.find_by(candidate_id: @matching_data['Apply candidate ID'])
-        .application_choices.find_by(course_option: course.course_options)
-        .status
+      ApplicationChoice.find_by(
```
Comment (duncanjbrown) [inline app/models/ucas_matched_application.rb:36]: You could do something like `ApplicationChoice.includes(:application_form).where(course_option: co, application_form: { candidate_id: c })`

**Comment by tvararu** [comment]
https://trello.com/c/RAdk0NeY/2301-fix-ucas-matching-sentry-error

**Review by duncanjbrown — APPROVED** [review]
LGTM!

**Pull request merged**